src/main/java/model/producer/BulldozerProducer.java

```diff
@@ -0,0 +1,20 @@
+package model.producer;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer {
```

Comment:
The `MachineProducer` interface should be parameterized to prevent creation of producers that do not produce `Machine` or its subtypes. Please update the interface implementation to `MachineProducer<Bulldozer>`.